Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -574,6 +574,32 @@ boost::shared_ptr<BankPulseTimes> LoadEventNexus::runLoadNexusLogs(
     if (run.hasProperty("period_log")) {
       auto *temp = run.getProperty("period_log");
       periodLog.reset(dynamic_cast<TimeSeriesProperty<int> *>(temp->clone()));
+
+      const auto nPeriodsInLog = periodLog->valuesAsVector().back();
+      if (nPeriodsInLog == 0 && nPeriods == 1) {
+        // Historic files can have 1 period but a period_log containing 0s
+        // if this is the case, "modernize" the local copy here
+        const int nValues = periodLog->realSize();
```

Comment:
Could we write this more concisely to avoid the temporaries as
```c++
periodLog->replaceValues(periodLog->timesAsVector(),
                         std::vector<int> newValues(periodLog->realSize(), 1));
```
?

Reply:
Good idea, will do this